src-tauri/src/main.rs

```diff
@@ -85,66 +82,70 @@ async fn main() {
         .and_then(|s| log::LevelFilter::from_str(s).ok())
         .unwrap_or(log::LevelFilter::Info);
 
-    wooting_macro_backend::MacroBackend::generate_directories();
+    MacroBackend::generate_directories().expect("unable to generate config directories");
 
     let backend = MacroBackend::default();
 
-    info!("Running the macro backend");
-
     #[cfg(any(target_os = "windows", target_os = "linux"))]
-    backend.init().await;
+    // The backend is run only on Windows and Linux, on macOS it won't work.
+    info!("Running the macro backend");
+    if let Err(e) = backend.init().await {
```

Comment:
Tried using -> ! in the init function but that didn't work well. Kept this way of handling the error for now (most likely will be reworked in 1.2 with the bigger backend rewrite)